react/features/base/participants/components/AvatarImage.native.js

```diff
@@ -0,0 +1,200 @@
+     *
+     * @inheritdoc
+     * @param {Object} nextProps - The read-only React Component props that this
+     * instance will receive.
+     * @returns {void}
+     */
+    componentWillReceiveProps(nextProps) {
+        const prevURI = this.props.source && this.props.source.uri;
+        const nextURI = nextProps.source && nextProps.source.uri;
+
+        if (typeof prevURI === 'undefined') {
+            clearTimeout(this._timeout);
+            if (typeof nextURI === 'undefined') {
+                this._timeout = setTimeout(() => {
+                    this.setState({ showDefault: true });
+                }, UNDEFINED_AVATAR_TIMEOUT);
+            } else {
+                this.setState({ showDefault: nextProps.forceDefault });
+            }
+        }
+    }
+
+    /**
+     * Clear the timer just in case. See {@code componentWillReceiveProps} for
+     * details.
+     *
+     * @inheritdoc
+     */
+    componentWillUnmount() {
+        clearTimeout(this._timeout);
+    }
+
+    /**
+     * Computes a hash over the URI and returns a HSL background color. We use
+     * 75% as lightness, for nice pastel style colors.
+     *
+     * @returns {string} - The HSL CSS property.
+     * @private
+     */
+    _getBackgroundColor() {
+        const uri = this.props.source.uri;
+        let hash = 0;
+
+        // If we have no URI yet we have no data to hash from, so use a random
+        // value.
+        if (typeof uri === 'undefined') {
+            hash = Math.floor(Math.random() * 360);
+        } else {
+            /* eslint-disable no-bitwise */
+
+            for (let i = 0; i < uri.length; i++) {
+                hash = uri.charCodeAt(i) + ((hash << 5) - hash);
+                hash |= 0;  // Convert to 32bit integer
+            }
+
+            /* eslint-enable no-bitwise */
+        }
+
+        return `hsl(${hash % 360}, 100%, 75%)`;
+    }
+
+    /**
+     * Error handler for image loading. When an image fails to load we'll mark
+     * it as failed and load the default URI instead.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onError() {
+        this.setState({ failed: true });
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     */
+    render() {
+        // eslint-disable-next-line no-unused-vars
+        const { forceDefault, source, style, ...props } = this.props;
+        const { failed, showDefault } = this.state;
+
+        if (failed || showDefault) {
+            const coloredBackground = {
+                ...style,
+                backgroundColor: this._getBackgroundColor(),
+                overflow: 'hidden'
+            };
+
+            let element = React.createElement(Image, {
+                ...props,
+                source: DEFAULT_AVATAR,
+                style: Platform.OS === 'android' ? style : coloredBackground
+            });
+
+            if (Platform.OS === 'android') {
```

Comment:
Well, the decision to apply the workaround was already taken above so I personally don't like that we have to make our mind up again. To me at this point in the execution creating a new `View` is necessary because `element` didn't take on `coloredBackground` as its `style` and we need someone else to take on it. But I can see how people will disagree with me. In which case I'd say that isolating the check in one place would be an improvement. For example:

```
const workaround3198 = Platform.OS === 'android';
...
style: workaround3198 ? style : coloredBackground;
...
if (workaround3198) {
...
```